### PR TITLE
Use SmallRye GraphQL Implementation

### PR DESCRIPTION
**Pull request by andymc12**
Part of #7956, this change updates the implementation to use SmallRye GraphQL.  It includes:
- updating to use the released MP GraphQL APIs
- changes to the non-TCK FAT test cases to account for updates in the spec/API/TCK, differences in the implementation, etc.
- moves the metrics integration into an auto-feature
- deletes the previous implementation
- updates the GraphQL-Java dependency to version 14.0 (latest stable release)

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
FYI The nlsprops file moved, but the text is unchanged since the last time it was approved.

**Comment by andrea-pichler** [comment]
Completed the requested L2 message review

**Comment by andymc12** [comment]
Some test failures in previous build.  Some are unrelated.  I think 1-2 failures may be because I changed "beta" to "noship" (originally thinking I would leave both implementations in place for a time).  I've changed it back to "beta" and will submit a new #build

File: dev/com.ibm.ws.com.graphql.java/bnd.bnd [hunk 1]
```diff
@@ -13,14 +13,14 @@
...
 
 -includeresource: \
-  @${repo;com.graphql-java:graphql-java;11.0},\
+  @${repo;com.graphql-java:graphql-java;14.0},\
   @${repo;com.graphql-java:java-dataloader;2.2.1},\
   @${repo;org.antlr:antlr4-runtime;4.7.1}
   
 Export-Package: \
-  graphql.*;version=11.0,\
+  graphql.*;version=13.0,\
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.com.graphql.java/bnd.bnd:25]: Should this be version 14?

File: dev/com.ibm.ws.io.smallrye.graphql/src/com/ibm/ws/io/smallrye/graphql/component/GraphQLExtension.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public void registerGraphQLBeans(@Observes AfterBeanDiscovery afterBeanDiscovery, BeanManager beanManager) {
+        ClassLoader tccl = getContextClassLoader();
+        Set<Bean<?>> beans = null;
+        synchronized (graphQLApiBeans) {
+            beans = graphQLApiBeans.get(tccl);
+        }
+        if (beans != null) {
+            for (Bean<?> bean : beans) {
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.finest("ANDY adding Bean, " + bean + " to afterBeanDiscovery");
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.io.smallrye.graphql/src/com/ibm/ws/io/smallrye/graphql/component/GraphQLExtension.java:83]: ANDY

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.defaultvalue/bootstrap.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.defaultvalue/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.basicMutation/bootstrap.properties [hunk 4]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.basicMutation/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.basicQuery/bootstrap.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.basicQuery/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.basicQueryWithConfig/bootstrap.properties [hunk 6]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.basicQueryWithConfig/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.deprecation/bootstrap.properties [hunk 7]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.deprecation/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.graphQLInterface/bootstrap.properties [hunk 8]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all:io.leangen.graphql.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.graphQLInterface/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.iface/bootstrap.properties [hunk 9]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.iface/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.inputFields/bootstrap.properties [hunk 10]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all:io.leangen.graphql.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.inputFields/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.metrics/bootstrap.properties [hunk 11]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.metrics/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.outputFields/bootstrap.properties [hunk 12]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all:io.leangen.graphql.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.outputFields/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.types/bootstrap.properties [hunk 13]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.types/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.voidQuery/bootstrap.properties [hunk 14]
```diff
@@ -1,4 +1,4 @@
-com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.*=all
+com.ibm.ws.logging.trace.specification=*=info:LogService=all:com.ibm.ws.microprofile.graphql.*=all:com.ibm.ws.io.smallrye.graphql.*=all:io.smallrye.graphql.*=all
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.voidQuery/bootstrap.properties:1]: do these traces currently need to be on? we should only enable them if we suspect there is an issue that we need to reproduce in order to limit how much we trace in the builds.

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/Variables.java [hunk 15]
```diff
@@ -0,0 +1,8 @@
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/Variables.java:1]: empty comment block, missing copyright header

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/VariablesAsString.java [hunk 16]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/VariablesAsString.java:1]: empty comment block, missing copyright header

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/VariablesIndividualProps.java [hunk 17]
```diff
@@ -0,0 +1,26 @@
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/VariablesIndividualProps.java:1]: empty comment block, missing copyright header

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/VariablesWeightOnly.java [hunk 18]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql.1.0_fat/test-applications/defaultvalueApp/src/mpGraphQL10/defaultvalue/VariablesWeightOnly.java:1]: empty comment block, missing copyright header

File: dev/com.ibm.ws.microprofile.graphql_fat_tck/publish/servers/FATServer/server.xml [hunk 19]
```diff
@@ -10,13 +10,17 @@
     <featureManager>
         <feature>componenttest-1.0</feature>
         <feature>localConnector-1.0</feature>
+        <!--
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.microprofile.graphql_fat_tck/publish/servers/FATServer/server.xml:13]: delete commented out features?

File: dev/com.ibm.ws.io.smallrye.graphql/resources/com/ibm/ws/io/smallrye/graphql/SmallRyeGraphQL.nlsprops [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+# This file has been modified to follow the WebSphere Message Guidelines.
+#
+# Briefly:
+#     MESSAGE_ID=<ComponentPrefix><MessageNumber><TypeCode>: <MessageText>
+#
+# The prefix for this component is:
+#     CWMGQ
+#
+# Message numbers are four digit numbers, from '0000' to '9999', inclusive.
+# Message numbers are filled to the left with '0', to a width of 4.
```
Comment (helyarp) [inline dev/com.ibm.ws.io.smallrye.graphql/resources/com/ibm/ws/io/smallrye/graphql/SmallRyeGraphQL.nlsprops:24]: I think the word "left" might be an accessibility violation (Avoid directional language). Maybe rephrase the "Message numbers are filled to the left with '0', to a width of 4." comment as "Message numbers start with '0' and are filled to a width of 4 digits."

File: dev/com.ibm.ws.io.smallrye.graphql/resources/com/ibm/ws/io/smallrye/graphql/SmallRyeGraphQL.nlsprops [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+# # {0} description of each insert field
+# MSG_DESCRIPTIVE_NAME_CWMGQ1xxxI=CWMGQ1xxxI: This is a message with inserts {0}
+# MSG_DESCRIPTIVE_NAME_CWMGQ1xxxI.explanation=Explanation text for the message
+# MSG_DESCRIPTIVE_NAME_CWMGQ1xxxI.useraction=User action text for the message
+#
+#-------------------------------------------------------------------------------
+
+ERROR_GENERATING_SCHEMA_CWMGQ0001E=CWMGQ0001E: An error in the {0} web module prevented the MicroProfile GraphQL application from starting.
+ERROR_GENERATING_SCHEMA_CWMGQ0001E.explanation=A severe error was detected while processing the MP GraphQL application in the specified module. This error prevents the application from starting successfully.
+ERROR_GENERATING_SCHEMA_CWMGQ0001E.useraction=Examine the logs to determine the error(s) in the GraphQL application, resolve them and redeploy the application.
```
Comment (helyarp) [inline dev/com.ibm.ws.io.smallrye.graphql/resources/com/ibm/ws/io/smallrye/graphql/SmallRyeGraphQL.nlsprops:57]: Change "error(s)" to "errors" and maybe rephrase the useraction: "Examine the logs to find the errors in the GraphQL application, resolve them, and then redeploy the application."

File: dev/com.ibm.ws.io.smallrye.graphql/src/com/ibm/ws/io/smallrye/graphql/component/GraphQLExtension.java (hunk 2 above)
Comment (andymc12) [reply dev/com.ibm.ws.io.smallrye.graphql/src/com/ibm/ws/io/smallrye/graphql/component/GraphQLExtension.java:83]: Can't a guy get a little credit for all his work?!  Sheesh!  ;-)  Good catch.

File: dev/com.ibm.ws.com.graphql.java/bnd.bnd (hunk 1 above)
Comment (andymc12) [reply dev/com.ibm.ws.com.graphql.java/bnd.bnd:25]: yep - will fix

File: dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.defaultvalue/bootstrap.properties (hunk 3 above)
Comment (andymc12) [reply dev/com.ibm.ws.microprofile.graphql.1.0_fat/publish/servers/mpGraphQL10.defaultvalue/bootstrap.properties:1]: I'd prefer to leave them on - this is new stuff and likely to break when tested on multiple platforms, etc - so having the trace on at first saves time with reproducing failures.  We can always turn it off later if people complain about it taking too much space.

File: dev/com.ibm.ws.io.smallrye.graphql/resources/com/ibm/ws/io/smallrye/graphql/SmallRyeGraphQL.nlsprops (hunk 21 above)
Comment (andymc12) [reply dev/com.ibm.ws.io.smallrye.graphql/resources/com/ibm/ws/io/smallrye/graphql/SmallRyeGraphQL.nlsprops]: sounds good.  I updated with both of your suggestions.  Thanks @helyarp !!

**Comment by andymc12** [comment]
Updated with fixes from code and message review.  New #build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Temporarily removing metrics support for GraphQL - this will be re-instated when we upgrade versions of SmallRye.  I'm still expecting a failure in the com.ibm.ws.anno bucket - but @tbitonti is planning to fix that in issue #11653 shortly - this PR will depend on his fix.

#build

**Comment by andymc12** [comment]
Rebased to include Tom's Jandex fix (#11686) - new #build

**Comment by andymc12** [comment]
All failures occur in other builds and seem completely unrelated to these changes.

**Comment by andymc12** [comment]
#run-libby-bot
